File: csharp/Svix/Models/OperationalWebhookEndpointSecretOut.cs
// this file is @generated
using System.Text;
using Newtonsoft.Json;

namespace Svix.Models
{
    public class OperationalWebhookEndpointSecretOut
    {
        [JsonProperty("key", Required = Required.Always)]
        public required string Key { get; set; }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();

            sb.Append("class OperationalWebhookEndpointSecretOut {\n");
            sb.Append("  Key: ").Append(Key).Append('\n');
            sb.Append("}\n");
            return sb.ToString();
        }
    }
}

File: ruby/lib/svix/models/operational_webhook_endpoint_secret_out.rb
# frozen_string_literal: true
# This file is @generated
require "json"

module Svix
  class OperationalWebhookEndpointSecretOut
    attr_accessor :key

    def initialize(attributes = {})
      unless attributes.is_a?(Hash)
        fail ArgumentError, "The input argument (attributes) must be a hash in `Svix::EndpointPatch` new method"
      end
      attributes.each do |k, v|
        instance_variable_set "@#{k}", v
      end
    end

    def self.deserialize(attributes = {})
      attributes = attributes.transform_keys(&:to_s)
      attrs = {
        'key': attributes["key"],
      }
      new attrs
    end

    def serialize
      out = Hash.new
      out["key"] = @key
      out.compact
    end

    # Serializes the object to a json string
    # @return String
    def to_json
      JSON.dump(serialize)
    end
  end
end
